common/router_data_model.go

```diff
@@ -998,33 +1701,106 @@ func CloneMap[V any](m cmap.ConcurrentMap[string, V]) cmap.ConcurrentMap[string,
 	return result
 }
 
-func (rdm *RouterDataModel) SubscribeToIdentityChanges(identityId string, subscriber IdentityEventSubscriber, isRouterIdentity bool) error {
+func (rdm *RouterDataModel) withIdentity(identityId string, f func(identity *Identity)) {
+	if identity, _ := rdm.Identities.Get(identityId); identity != nil {
+		f(identity)
+	}
+}
+
+func (rdm *RouterDataModel) withLockedIdentity(panicOnNil bool, identityId string, f func(identity *Identity)) {
+	if identity, _ := rdm.Identities.Get(identityId); identity != nil {
+		identity.lock.Lock()
+		defer identity.lock.Unlock()
+		f(identity)
+	} else if panicOnNil {
+		panic(fmt.Errorf("identity '%s' not found", identityId))
+	} else {
+		f(nil)
+	}
+}
+
+func (rdm *RouterDataModel) withServices(serviceIds []string, f func(service *Service)) {
+	for _, serviceId := range serviceIds {
+		if service, _ := rdm.Services.Get(serviceId); service != nil {
+			f(service)
+		}
+	}
+}
+
+func (rdm *RouterDataModel) withService(serviceId string, f func(service *Service)) {
+	if service, _ := rdm.Services.Get(serviceId); service != nil {
+		f(service)
+	}
+}
+
+func (rdm *RouterDataModel) withServicePolicy(servicePolicyId string, f func(service *ServicePolicy)) {
+	if servicePolicy, _ := rdm.ServicePolicies.Get(servicePolicyId); servicePolicy != nil {
+		f(servicePolicy)
+	}
+}
+
+func (rdm *RouterDataModel) withPostureChecks(postureCheckIds []string, f func(PostureCheck *PostureCheck)) {
+	for _, postureCheckId := range postureCheckIds {
+		if postureCheck, _ := rdm.PostureChecks.Get(postureCheckId); postureCheck != nil {
+			f(postureCheck)
+		}
+	}
+}
+
+func (rdm *RouterDataModel) withPostureCheck(postureCheckId string, f func(PostureCheck *PostureCheck)) {
+	if postureCheck, _ := rdm.PostureChecks.Get(postureCheckId); postureCheck != nil {
+		f(postureCheck)
+	}
+}
+
+func (rdm *RouterDataModel) SubscribeToIdentityChanges(identityId string, subscriber IdentityEventSubscriber, isRecreatableIdentity bool) error {
```

Comment:
Why the change from `isRouterIdentity` to `isRecreatableIdentity`?

I know what a `router identity` is, but a `recreatable identity` is not something I'm familiar with. Is this a new concept? Does it go beyond router identities now? Most use cases in here are `false`. I can only find `true` use cases in older code that still references the router identity as the case.